server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex

```diff
@@ -1,12 +1,76 @@
 defmodule HeliosWeb.WebHooks.SlackController do
   use HeliosWeb, :controller
-  alias Helios.{Repo, Event}
+  require Logger
+  alias Helios.{Repo, Event, SlackChannelNames}
+
+  defp insert_channel_id(channel_id) do
+    unless SlackChannelNames
+           |> SlackChannelNames.with_id(channel_id)
+           |> Repo.exists?() do
+      try do
+        body = %{
+          "channel" => channel_id
+        }
+
+        request_body = URI.encode_query(body)
+
+        headers = [
+          {"Accept", "application/json"},
+          {"Authorization",
+           "Bearer #{Application.get_env(:helios, HeliosWeb.Endpoint)[:slack_bearer_token]}"},
+          {"Content-Type", "application/x-www-form-urlencoded"}
+        ]
+
+        channel_response =
+          HTTPoison.post!("https://slack.com/api/conversations.info", request_body, headers,
+            follow_redirect: true
+          )
+
+        json = Jason.decode!(channel_response.body)
+        name = json["channel"]["name"]
+
+        Repo.insert!(%SlackChannelNames{
+          channel_id: channel_id,
+          channel_name: name
+        })
+      rescue
+        e -> IO.inspect(e)
```

Comment:
This isn't really recovering from an error, it's just logging it and going on as though nothing went wrong. It's also spitting it out to stdout instead of properly logging it which may or may not get dropped in production depending on how the logs are captured and stored.

If we don't know how to programmatically recover from an error the best thing to do is probably to let it propagate up to the API layer and (hopefully) return a meaningful error message to the client. That way the user knows something broke and can react appropriately.